clients/feeder/transaction.go

```diff
@@ -44,22 +45,72 @@ func (fs *FinalityStatus) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+type TransactionType uint8
+
+const (
+	Invalid TransactionType = iota
+	TxnDeclare
+	TxnDeploy
+	TxnDeployAccount
+	TxnInvoke
+	TxnL1Handler
+)
+
+func (t TransactionType) String() string {
+	switch t {
+	case TxnDeclare:
+		return "DECLARE"
+	case TxnDeploy:
+		return "DEPLOY"
+	case TxnDeployAccount:
+		return "DEPLOY_ACCOUNT"
+	case TxnInvoke:
+		return "INVOKE_FUNCTION"
+	case TxnL1Handler:
+		return "L1_HANDLER"
+	default:
+		return "<unknown>"
+	}
+}
+
+func (t TransactionType) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("%q", t.String())), nil
```

Comment:
in case of %s, %v and %q it's safe to pass just t, without calling `.String()` explicitly

Reply:
Done